tests/test_models.py

```diff
@@ -662,6 +664,14 @@ def test_creation_time(self, db_session):
         self.add(db_session, node)
         assert node.creation_time is not None
 
+    def test_details(self, db_session):
+        net = models.Network()
+        db_session.add(net)
+        node = models.Node(network=net)
+        node.details = {"my_data": [1, 2, 3]}
+        self.add(db_session, node)
+        assert tuple(node.details["my_data"]) == (1, 2, 3)
+
```

Comment:
Pretty sure you could write this test as:
```
    def test_details(self, a):
        node = a.node()
        node.details = {"my_data": [1, 2, 3]}
        assert tuple(node.details["my_data"]) == (1, 2, 3)
```

Reply:
Ah, I see... that entire module still does all the fixture created "by hand", so maybe consistency is better.

Reply:
Without adding the node to the session after setting the value you're not actually verifying that the data is being validated, written and read by the ORM. The test setup in that module definitely isn't ideal, but I don't think this is the time to refactor it.

Reply:
The `a` fixture calls `db_session.add()` and `db_session.flush()`, so I think it would be equivalent, but I agree it would be best to hit that entire module at once (and at some other time).